### PR TITLE
Implement correct calculating of rotation spread

### PR DESCRIPTION
**Pull request by yklishevich**

### CODE REVIEW

File: FantasmoSDK/Classes/Extensions/Array+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//
+//  Array+Extension.swift
+//  FantasmoSDK
+//
+//  Created by Yauheni Klishevich on 20.07.2021.
+//
+
+import Foundation
+
+extension Array {
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Extensions/Array+Extension.swift:10]: Let's not use extensions for syntactic sugar. Also, `!` is already a very clear, common, and succinct way to express negation.

File: FantasmoSDK/Classes/Models/Analytics/Impl/RotationAngleCalculator.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            if diff > 0 {
+                return diff - 2 * .pi // returns diff in [-2pi, 0]
+            }
+            else {
+                return diff
+            }
+        }
+    }
+    
+    private func probabilityOfPossitiveDiffDeducedFromPreviousLimitedAngles(for nextOrientationAngle: Float) -> Float {
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Models/Analytics/Impl/RotationAngleCalculator.swift:74]: This makes me think we should really be working in quaternions and not Euler angles. Wouldn't that remove all of the sudden jumps at angle extrema?

File: FantasmoSDK/Classes/Utilities/ObservationToken.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+//
+//  Created by Yauheni Klishevich on 19.07.2021.
+//
+
+import Foundation
+
+/// Token that is used in the same way as `AnyCancellable` is used in `Combine` framework.
+/// Functions for adding closure based observers may return this token to allow unsubscribing.
+/// The necessity of the class was driven by accebility of `Combine` framework starting from only iOS 13.0, while SDK deploment target was lower.
+public class ObservationToken {
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Utilities/ObservationToken.swift:13]: Seems that we need to merge `develop` into this branch. There are a bunch of changes showing up from the other PR.

**Review by lucaskuzma — CHANGES_REQUESTED** [review]

File: FantasmoSDK/Classes/Models/Analytics/Impl/RotationAngleCalculator.swift (hunk 2 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Models/Analytics/Impl/RotationAngleCalculator.swift:74]: I don't think that quaternions  make little difference. We will have to convert quaternion into Euler anyway as we need to calculate the spread of Euler angles. Quaternion represents the orientation of the axis and rotation angle about this axis. Each frame will have different axis orientations and different angles. It's not clear how we can define quaternion based spread

**Pull request merged**